### PR TITLE
Ensure namespace existence is verified and forget key if not found

### PR DESCRIPTION
**Pull request by Barakmor1**
we should verify ns exist before we start the reconciliation loop to prevent infinite key reenqueueing.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/aaq-controller/aaq-gate-controller/aaq-gate-controller_test.go [hunk 1]
```diff
@@ -44,14 +44,23 @@ var _ = Describe("Test aaq-gate-controller", func() {
...
+			namespaceLister := testsutils.FakeNamespaceLister{Namespaces: map[string]*corev1.Namespace{testNs: {ObjectMeta: metav1.ObjectMeta{Name: testNs}}}}
 			recorder := record.NewFakeRecorder(100)
-			qc := setupAAQGateController(cli, nil, nil, aaqjcInformer, recorder)
+			qc := setupAAQGateController(cli, nil, nil, aaqjcInformer, namespaceLister, recorder)
 			err, es := qc.execute(testNs)
 			Expect(err).ToNot(HaveOccurred())
 			Expect(es).To(Equal(Forget))
 		})
 
+		It("the key namespace doesn't exist should forget key", func() {
```
Comment (vladikr) [inline pkg/aaq-controller/aaq-gate-controller/aaq-gate-controller_test.go:55]: maybe reword this to something like "should forget the key if its namespace doesn't exist" ?

**Comment by vladikr** [comment]
Great. Looks good.
/approve

File: pkg/aaq-controller/additional-cluster-quota-controllers/acrq-controller/acrq-controller_test.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	testsutils "kubevirt.io/application-aware-quota/pkg/tests-utils"
+)
+
+var _ = Describe("Test aaq-gate-controller", func() {
+	var testNs = "test"
+
+	Context("Test execute when ", func() {
+		It("the key namespace doesn't exist should forget key", func() {
```
[the remark this answers is not in the document]
Comment (Barakmor1) [reply pkg/aaq-controller/additional-cluster-quota-controllers/acrq-controller/acrq-controller_test.go]: Done :)

**Comment by vladikr** [comment]
/lgtm
Thanks!
